base/src/main/scala/data/DisjunctionFunctions.scala

```diff
@@ -1,10 +1,17 @@
 package scalaz
 package data
 
+import Disjunction.{\/, \/-, -\/}
+
 trait DisjunctionFunctions {
-  @inline def -\/[L](value: L): Disjunction[L, Nothing] = Disjunction.-\/(value)
-  @inline def \/-[R](value: R): Disjunction[Nothing, R] = Disjunction.\/-(value)
+  @inline def -\/[L](value: L): Disjunction[L, Nothing] = -\/(value)
```

Comment:
I think it was initially invariant, but after discussing with @S11001001 and/or @vmarquez I got convinced it should be covariant.

I can't remember the details, but let's now settle on something and I'll add it in `CONTRIBUTING`.

Do we want to allow covariance on datatype? what was the strong reason in favor of it?

My initial thinking was that if the user always create proper constructor for it's ADT, there is no need for having variance on datatype. That is quite a cost for the user though.

Once we agree as a general rule, I'll update `CONTRIBUTING` and `Maybe` (which is currently invarriant) accordingly.